webapp/src/components/admin_console_settings/app_id/index.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {EVENT_APP_INPUT_CHANGE} from '../constants';
+
+interface Props {
+    id: string;
+    label: string;
+    helpText?: React.ReactNode;
+    placeholder?: string;
+    value: string;
+    disabled?: boolean;
+    onChange: (id: string, value: string) => void;
+}
+
+// Component for setting the app ID
+const AppID: React.FC<Props> = (props) => {
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const newValue = e.target.value;
+        props.onChange(props.id, newValue);
+
+        // Dispatch custom event for real-time validation
+        window.dispatchEvent(new CustomEvent(EVENT_APP_INPUT_CHANGE, {
```

Comment:
Shouldn't we propagate this to the parent component and let the parent component report the Manifest download button? Unsure about the limitations in custom admin webapp components.

Also, should we prefix the `EVENT_APP_INPUT_CHANGE` with the plugin ID? Since any other system console javascript could be listening to this.

Reply:
I added the plugin id to the event name in the follow-up PR here:  https://github.com/mattermost/mattermost-plugin-msteams-devsecops/pull/57

As for propagating to the parent and managing the Download button state that way, I did try that initially and found it to be much more complex. In one iteration I used redux like the Calls plugin does, and the logic for simply changing state of one component based on values of others was spread out of many files and was hard to debug.  IMO this is a simple solution albeit a bit old-school.